### PR TITLE
LPS-132871 ClassicEditor content is missing

### PR DESCRIPTION
**Pull request by markocikos**
Fixes: https://issues.liferay.com/browse/LPS-132871

@boton @julien 

This is a bugfix, but basically a followup on https://github.com/liferay-frontend/liferay-portal/pull/954. The bug was caused by https://github.com/liferay-frontend/liferay-portal/pull/954/commits/84432f7ed93f4c474f22b44f48548f1ba6769dd7, merged as a part of that PR. Looking at the discussion in the PR, I see you wanted to use `data` prop, [expecting](https://github.com/liferay-frontend/liferay-portal/pull/954#issuecomment-839672548) that React component does the updating. But, you were looking at `ckeditor4-react` master and we are using an older version of React component in DXP. If you tried it, we would show unformatted code (see 1.3.0 gif).

In this PR, we are updating `ckeditor4-react` to the latest version and using `data` param to set content.

**Important note** @boton : With this PR, it is likely that we will have regression on  https://issues.liferay.com/browse/LPS-130112. `onChange` is now be triggered internally and we cannot avoid it. I am not sure now to replicate the issue in that ticket, or if it's still valid. We should probably find a different way to work around, because we don't want to be stuck with old version of `ckeditor4-react` forever. Can you please look into this?

Before PR:
before

After using `data` prop, without `ckeditor4-react` update:
after1

After this PR:
after

### CODE REVIEW

**Comment by boton** [comment]
Hey, @markocikos.

With this [commit](https://github.com/liferay-frontend/liferay-portal/pull/1133/commits/1527d266dfe25bd1d75de8ba5a64016617eb13dc) shouldn’t have regression bug :D 

I´ll take a look when is merged, thanks a lot for keeping this https://issues.liferay.com/browse/LPS-130112 in mind.

**Comment by markocikos** [comment]
> With this commit we shouldn’t have regression bug

@boton Have in mind, this PR is not just that commit. We are updating `ckeditor4-react` to latest version, and it will now do the exact thing that caused the bug. We will again be [invoking setData on  'instanceReady'](https://github.com/ckeditor/ckeditor4-react/blob/v1.4.2/src/ckeditor.jsx#L89-L95). The effect may be the same as reverting https://github.com/liferay-frontend/liferay-portal/commit/84432f7ed93f4c474f22b44f48548f1ba6769dd7.

File: modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/ClassicEditor.js [hunk 1]
```diff
@@ -123,6 +123,7 @@ const ClassicEditor = React.forwardRef(
 				<Editor
 					className="lfr-editable"
 					config={getConfig()}
+					data={contents}
```
Comment (julien) [inline modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/ClassicEditor.js:126]: Thanks @markocikos :+1:

**Comment by markocikos** [comment]
@boton @julien  Ok, let's merge this. The change is required to fix the bug. If there is a regression in LPS-130112, the path forward should be to find a different workaround for it.

**Comment by markocikos** [comment]
ci:forward

**Comment by boton** [comment]
I run a locally Poshi test that points out this bug https://github.com/liferay-lima/liferay-portal/pull/1641#issuecomment-806571034 and anything blows up.

**Comment by boton** [comment]
The Friday @Jeremy-Ch (Lima QA member) alerts me that this bug is being reproducible again, only in Poshi tests. We are trying to update the tests because they aren't reproducible in UI but can give us problems in the future.

/cc @markocikos @julien

**Comment by julien** [comment]
> We are trying to update the tests because they aren't reproducible in UI but can give us problems in the future.

That's interesting, could you explain what the poshy test is doing and why it isn't reproducible by hand?
When you say "can give us problems in the future?" what problems are we talking about? 
Let's talk about it when you can.

**Comment by boton** [comment]
> That's interesting, could you explain what the poshy test is doing and why it isn't reproducible by hand?

Because if the original data is created via API in plain text without HTML format (like in the Poshi test) the CKEditor will parse the data and call the `onChange`.

You can check a test case in boton#224.

> When you say "can give us problems in the future?" what problems are we talking about?

If we are thrusting in the `onChange` to track changes, the above case could be a problem.

We are tracking dirty state based on onChange callbacks in Translation for example.

image

**Comment by julien** [comment]
Hi @boton ,

> Because if the original data is created via API in plain text without HTML format (like in the Poshi test) the CKEditor will parse the data and call the onChange.

Shouldn't we update the test then? Shouldn't tests be "testing" what is actually happening instead of something else?

> If we are thrusting in the onChange to track changes, the above case could be a problem.
>  We are tracking dirty state based on onChange callbacks in Translation for example.

I don't understand what you mean sorry. Are you saying that this [`onChangeCallback`](https://github.com/liferay/liferay-portal/blob/a0de1166dd81ef9cfd22f8c3cae1066483049425/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/ClassicEditor.js#L68-L85) funciton is working incorrectly? (I wouldn't be surprised)

**Comment by boton** [comment]
> Shouldn't we update the test then? Shouldn't tests be "testing" what is actually happening instead of something else?

Yes, @Jeremy-Ch updated the test to avoid this problem

> I don't understand what you mean sorry. Are you saying that this onChangeCallback function is working incorrectly? (I wouldn't be surprised)

I expect the onChange method only is going to be called when the final user makes a real change, not in the initialization phase parsing text to HTML. I'm not sure if the problem is in the `onChangeCallback` but maybe.

This isn't a real problem for now when it becomes one we will have to look at it :D 

Thanks @julien

**Comment by julien** [comment]
Ok thanks @boton. If ever we have to look into this, I'll be happy to help.

**Pull request closed without merging**